### PR TITLE
operator: iaa: Drop hardcoded deployment, use embedded YAML

### PR DESCRIPTION
**Pull request by ozhuraki**

### CODE REVIEW

File: pkg/controllers/iaa/controller.go [hunk 1]
```diff
@@ -188,8 +188,7 @@ func (c *controller) NewDaemonSet(rawObj client.Object) *apps.DaemonSet {
 	yes := true
 	daemonSet := apps.DaemonSet{
 		ObjectMeta: metav1.ObjectMeta{
-			//Namespace:    devicePlugin.Namespace,
-			Namespace:    metav1.NamespaceSystem,
+			Namespace:    c.ns,
```
Comment (mythi) [inline pkg/controllers/iaa/controller.go:191]: this whole function needs be updated to what other controllers have:
https://github.com/intel/intel-device-plugins-for-kubernetes/blob/959cef517f620ff0ecc523e250d89056fe916733/pkg/controllers/dsa/controller.go#L174-L181

**Review by mythi — CHANGES_REQUESTED** [review]

**Comment by ozhuraki** [comment]
@mythi 
Thanks! Fixed, please take a look

**Comment by mythi** [comment]
@ozhuraki thanks. looks like to commit message is still a bit outdated

File: pkg/controllers/iaa/controller.go [hunk 2]
```diff
@@ -170,131 +169,22 @@ func addInitContainer(ds *apps.DaemonSet, dp *devicepluginv1.IaaDevicePlugin) {
 func (c *controller) NewDaemonSet(rawObj client.Object) *apps.DaemonSet {
 	devicePlugin := rawObj.(*devicepluginv1.IaaDevicePlugin)
 
-	var nodeSelector map[string]string
+	daemonSet := deployments.DSAPluginDaemonSet()
```
Comment (hj-johannes-lee) [inline pkg/controllers/iaa/controller.go:172]: Maybe typo here..!

**Review by hj-johannes-lee — CHANGES_REQUESTED** [review]

**Comment by ozhuraki** [comment]
@mythi @hj-johannes-lee 
Thanks! Fixed.

**Review by mythi — APPROVED** [review]

**Comment by mythi** [comment]
> @mythi @hj-johannes-lee Thanks! Fixed.

@hj-johannes-lee can you approve to unblock merging

**Review by hj-johannes-lee — APPROVED** [review]

**Pull request merged**